src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useRef, useState } from 'react';
+import { Person } from '../types/Person';
+/* eslint-disable prettier/prettier */
+/* eslint-disable max-len */
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  debounceDelay: number;
+}
+
+export const Autocomplete = ({
+  people,
+  onSelected,
+  debounceDelay,
+}: AutocompleteProps) => {
+  const [searchInput, setSearchInput] = useState('');
+  const [isSearchInputFocus, setIsSearchInputFocus] = useState(false);
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>([]);
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const debounceTimeout = setTimeout(() => {
+      const result = searchInput
+        ? people.filter(person =>
+          person.name.toLowerCase().includes(searchInput.toLowerCase()),
+        )
+        : people;
```

Comment:
The filtering logic should not be triggered by spaces alone. Consider trimming the `searchInput` or checking if it contains non-space characters before applying the filter.